tests/test_metadata.py

```diff
@@ -34,3 +25,29 @@ def test_metadata() -> None:
     source_unit.remove_metadata()
     assert with_metadata != source_unit.bytecode_init("Dai")
     assert source_unit.bytecode_init("Dai") == DAI_BYTECODE
+
+
+def test_metadata() -> None:
+    """
+    Test the metadata
+
+    Returns:
+
+    """
+    crytic_compile_instance = CryticCompile(
+        "0x6B175474E89094C44Da98b954EedeAC495271d0F",  # solc 0.5.12
+    )
+
+    metadata_checks(crytic_compile_instance)
+
+
+def test_metadata_deferred() -> None:
+    crytic_compile_instance = CryticCompile(
+        "0x6B175474E89094C44Da98b954EedeAC495271d0F",  # solc 0.5.12
+        crytic_defer_compilation="true",
```

Comment:
```suggestion
        crytic_defer_compilation=True,
```

Reply:
That how I had it initially, but I was getting mypy errors due to the type of kwargs (only takes str right now). If we are ok with changing that to something like `str | bool` then I'm all for changing it.